### DISCUSSION TITLE
KongIngress taking time to sync

### DISCUSSION THREAD

**Discussion by guptadeepali39** [Kong/kong#8367] [Help]
Hi,

I am using DB-less kong and I am creating a kongIngress resource as below:

apiVersion: configuration.konghq.com/v1
kind: KongIngress
metadata:
    name: demo-health-checking
upstream:
  healthchecks:
    active:
      healthy:
        interval: 2
        successes: 3
      http_path: /SEG2/test
      type: http
      unhealthy:
        http_failures: 1
        interval: 5
    passive:
      healthy:
        successes: 3
      unhealthy:
        http_failures: 3

And then patching this on my service, but the issue here is it is taking 1 hour to sync.
Can I know, how can I fix this.

**Comment by hbagdi** [comment]
cc @rainest @shaneutt

**Comment by rainest** [comment]
What controller version are you using, and can you upload a copy of logs with `CONTROLLER_LOG_LEVEL=debug` set in the controller container environment?

As of 2.x you should see both the initial read of the KongIngress/Service logged, and then after the time of the next sync, with any errors that occurred.

2.0 had a known issue where the sync process did not properly report errors on timeouts. This could appear as resources taking a long time to sync because most syncs failed to complete and terminated before that particular resource would have synced, but eventually you'll get a lucky sync that does complete in time.

2.1 fixes this issue, though note that it only fixes the lack of error reporting. If you upgrade and do start to see timeout errors, you'll need to set CONTROLLER_PROXY_TIMEOUT_SECONDS to a higher value (the default is `10` seconds).

**Comment by guptadeepali39** [comment]
@rainest @hbagdi I was using kong version 2.6 and ingress controller version 2.0.
After changing the kong version to 2.7 and updating ingress controller version 2.1, I observed that syncs are happening immediately.
So, updating the version solved my problem.
Thanks a lot.

**Reply by rainest** [reply]
2.1 shouldn't actually cause any change on its own to behavior; it will only alert you to note that timeouts are occurring, and that you need to raise the timeout. I'm not aware of anything else obvious in the changeset that would account for fixing the problem.

It's possible there was some other service degradation that slowed down config pushes over time, so keep an eye on logs if the problem reoccurs.

**Comment by guptadeepali39** [comment]
@rainest I am also not sure, because right now just updated both versions and it started syncing immediately. I will keep on observing logs, I will post if I see any issue further

**Reply by guptadeepali39** [reply]
@rainest I noticed a behavior, when I create the KongIngress and patch it to service, it starts hitting the active health check endpoint say  /SEG2/test in my scenario. But after 10-15 mins it stops hitting it and after 3 consecutive 500 response making the service down and it is not getting up, as it is now not hitting the active end point

**Comment by guptadeepali39** [comment]
@rainest @hbagdi so putting the health checks with kong ingress are not working. It works initially but stop working after sometime.
It starts working when restart kong. But that is not the solution.
Is there any other option with which we can setup health checks and which is production ready solution.

**Reply by rainest** [reply]
Are you seeing the upstream healthcheck configuration disappear after a period of time, or is it still the desired configuration, just not checking as you'd like? The controller is responsible for putting the configuration in place, but after that the health state and checks are up to the Kong proxy code.

Issues clearing upon restart suggest that it's something wrong with the checker state. When problems occur, what is the [health state of the upstream](https://docs.konghq.com/gateway/2.7.x/admin-api/#show-upstream-health-for-node), and do you see the active check requests logged by the upstream application?

**Reply by guptadeepali39** [reply]
@rainest so health check configuration remains there. 
What happening is for the active check I have configured a get endpoint in my application and have put on logs for debugging , so initially it hits it after every second(I have set to 2 seconds in kong ingress resource) and health check works as expected.
But after some time, it stops hitting the endpoint, and at that time my upstream is in active state.
Now if my upstream goes down because of passive health checks, it never gets back into active state because now it is no more hitting the active point.
